src/ProcessManager.php

```diff
@@ -530,12 +530,13 @@ public function run()
         ob_implicit_flush(1);
 
         $this->loop = Factory::create();
-        $this->controller = new UnixServer($this->getControllerSocketPath(), $this->loop);
-        $this->controller->on('connection', [$this, 'onSlaveConnection']);
 
         $this->web = new Server(sprintf('%s:%d', $this->host, $this->port), $this->loop, ['backlog' => self::TCP_BACKLOG]);
         $this->web->on('connection', [$this, 'onRequest']);
 
+        $this->controller = new UnixServer($this->getControllerSocketPath(), $this->loop);
```

Comment:
Moving this lines will allow the main public socket to fail if it's already occupied before creating the controller socket